BUG: v1.4.3 - ${GROUP} interpolation in service stanza broken

### Nomad version
Nomad v1.4.3 (f464aca721d222ae9c1f3df643b3c3aaa20e2da7)

### Operating system and Environment details
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
Environment: Windows 11, WSL2

### Issue
The `$GROUP` variable is not interpolated in service stanza when using `provider = "nomad"`
https://developer.hashicorp.com/nomad/docs/job-specification/service#name

### Reproduction steps
```
job "frontend" {
  datacenters = ["dc1"]
  type = "service"

  group "webserver" {
    count = 1

    network {
      port "http" {
        to = 80
      }
      port "admin" {
        to = 8080
      }
    }

    service {
      name = "${GROUP}"
      port = "http"
      tags = ["http"]
      provider = "nomad"
    }

    task "nginx" {
      driver = "docker"
      config {
        image = "nginx:latest"
        ports = ["http"]
      }

      service {
        name = "${GROUP}-${TASK}"
        port = "admin"
        tags = ["admin"]
        provider = "nomad"
      }

      resources {
        cpu = 1000
        memory = 512
      }
    }

  }
}
```

#### Expected Result
Context: Prometheus `nomad_sd_configs` (Nomad service discovery for Prometheus)

**group level**
```
__meta_nomad_service="webserver"
__meta_nomad_service_id="_nomad-task-9df05b85-60ac-1dae-1851-8bebf35067fd-group-webserver-webserver-http"
```

**task level**
```
__meta_nomad_service="webserver-nginx"
__meta_nomad_service_id="_nomad-task-9df05b85-60ac-1dae-1851-8bebf35067fd-nginx-webserver-nginx-metrics"
```

#### Actual Result
**NOMAD LOG**
```
2022-12-31T21:59:56.145+0100 [ERROR] client.alloc_runner.task_runner: prestart failed: alloc_id=a6d73815-c199-f0ff-200d-cc780d3eddca task=nginx error="prestart hook \"validate\" failed: service (${GROUP}-metrics) failed validation: Service name must be valid per RFC 1123 and can contain only alphanumeric characters or dashes and must be no longer than 63 characters"
2022-12-31T21:59:56.146+0100 [INFO]  client.alloc_runner.task_runner: not restarting task: alloc_id=a6d73815-c199-f0ff-200d-cc780d3eddca task=nginx reason="Error was unrecoverable"
```

Context: Prometheus `nomad_sd_configs` (Nomad service discovery for Prometheus)
**group level**
```
__meta_nomad_service="${GROUP}"
__meta_nomad_service_id="_nomad-task-9df05b85-60ac-1dae-1851-8bebf35067fd-group-webserver-${GROUP}-http"
```

**task level**
```
__meta_nomad_service="${GROUP}-nginx"
__meta_nomad_service_id="_nomad-task-9df05b85-60ac-1dae-1851-8bebf35067fd-nginx-${GROUP}-nginx-metrics"
```

--- a/website/content/docs/job-specification/service.mdx
+++ b/website/content/docs/job-specification/service.mdx
@@ -86,9 +86,9 @@ Connect][connect] integration.
 - `connect` - Configures the [Consul Connect][connect] integration. Only
   available on group services and where `provider = "consul"`.
 
-- `name` `(string: "<job>-<group>-<task>")` - Specifies the name this service
+- `name` `(string: "<job>-<taskgroup>-<task>")` - Specifies the name this service
   will be advertised as in Consul. If not supplied, this will default to the
-  name of the job, group, and task concatenated together with a dash, like
+  name of the job, task group, and task concatenated together with a dash, like
   `"docs-example-server"`. Each service must have a unique name within the
   cluster. Names must adhere to [RFC-1123
   §2.1](https://tools.ietf.org/html/rfc1123#section-2) and are limited to
@@ -99,9 +99,9 @@ Connect][connect] integration.
   following keys are also available:
 
   - `${JOB}` - the name of the job
-  - `${GROUP}` - the name of the group
+  - `${TASKGROUP}` - the name of the task group
   - `${TASK}` - the name of the task
-  - `${BASE}` - shorthand for `${JOB}-${GROUP}-${TASK}`
+  - `${BASE}` - shorthand for `${JOB}-${TASKGROUP}-${TASK}`
 
   Validation of the name occurs in two parts. When the job is registered, an initial validation pass checks that
   the service name adheres to RFC-1123 §2.1 and the length limit, excluding any variables requiring interpolation.

